pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -66,8 +72,31 @@
   @Inject
   AccessControlFactory _accessControlFactory;
 
+  @POST
+  @Path("queryByFormat")
+  public String postQuery(String requestJsonStr, @Context HttpHeaders httpHeaders) {
+    try {
+      JsonNode requestJson = JsonUtils.stringToJsonNode(requestJsonStr);
+      String query = requestJson.get("query").asText();
+      String traceEnabled = "false";
+      if (requestJson.has("trace")) {
+        traceEnabled = requestJson.get("trace").toString();
+      }
+      String queryFormat = "pql";
+      if (requestJson.has("queryFormat")) {
+        queryFormat = requestJson.get("queryFormat").asText();
+      }
+      LOGGER.debug("Trace: {}, Running query: {}", traceEnabled, query);
+      return getQueryResponse(query, traceEnabled, httpHeaders, queryFormat);
+    } catch (Exception e) {
+      LOGGER.error("Caught exception while processing post request", e);
+      return QueryException.getException(QueryException.INTERNAL_ERROR, e).toString();
+    }
+  }
+
   @POST
   @Path("pql")
+  @Deprecated
```

Comment:
Another backward compatible way of making this change is to keep "pql" path for PQL queries 'as-is' and add a "sql" path for SQL queries. Unclear to me which is a better option at the moment, but stating so that others can also pitch in. At least, in this approach there is no need to deprecate existing api.